feature/qos/otg_tests/egress_strict_priority_scheduler_test/egress_strict_priority_scheduler_test.go

```diff
@@ -0,0 +1,2576 @@
+	schedulerIntfs := []struct {
+		desc      string
+		queueName string
+		scheduler string
+	}{{
+		desc:      "output-interface-BE1",
+		queueName: queues.BE1,
+		scheduler: "scheduler",
+	}, {
+		desc:      "output-interface-AF1",
+		queueName: queues.AF1,
+		scheduler: "scheduler",
+	}, {
+		desc:      "output-interface-AF2",
+		queueName: queues.AF2,
+		scheduler: "scheduler",
+	}, {
+		desc:      "output-interface-AF3",
+		queueName: queues.AF3,
+		scheduler: "scheduler",
+	}, {
+		desc:      "output-interface-AF4",
+		queueName: queues.AF4,
+		scheduler: "scheduler",
+	}, {
+		desc:      "output-interface-NC1",
+		queueName: queues.NC1,
+		scheduler: "scheduler",
+	}}
+
+	t.Logf("QoS output interface config: %v", schedulerIntfs)
+	for _, tc := range schedulerIntfs {
+		i := q.GetOrCreateInterface(dp3.Name())
+		i.SetInterfaceId(dp3.Name())
+		i.GetOrCreateInterfaceRef().Interface = ygot.String(dp3.Name())
+		if deviations.InterfaceRefConfigUnsupported(dut) {
+			i.InterfaceRef = nil
+		}
+		output := i.GetOrCreateOutput()
+		schedulerPolicy := output.GetOrCreateSchedulerPolicy()
+		schedulerPolicy.SetName(tc.scheduler)
+		queue := output.GetOrCreateQueue(tc.queueName)
+		queue.SetName(tc.queueName)
+		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+	}
+}
+
+func runCliCommand(t *testing.T, dut *ondatra.DUTDevice, cliCommand string) string {
+	cliClient := dut.RawAPIs().CLI(t)
+	output, err := cliClient.RunCommand(context.Background(), cliCommand)
+	if err != nil {
+		t.Fatalf("Failed to execute CLI command '%s': %v", cliCommand, err)
+	}
+	t.Logf("Received from cli: %s", output.Output())
+	return output.Output()
+}
+
+func configureQoSGlobalParams(t *testing.T, dut *ondatra.DUTDevice) {
+
+	queues := netutil.CommonTrafficQueues(t, dut)
+	qosQNameSet := `
+	configure terminal
+	!
+	qos tx-queue %d name %s
+	!
+	`
+	qosMapTC := `
+	configure terminal
+	!
+	qos map traffic-class %d to tx-queue %d
+	!
+	`
+
+	qosCfgTargetGroup := `
+	configure terminal
+	!
+	qos traffic-class %d name %s
+	!
+	`
+
+	runCliCommand(t, dut, "show version")
+
+	qList := []string{queues.BE1, queues.AF1, queues.AF2, queues.AF3, queues.AF4, queues.NC1}
+	for index, queue := range qList {
+
+		runCliCommand(t, dut, fmt.Sprintf(qosQNameSet, index, queue))
+		time.Sleep(time.Second)
+		runCliCommand(t, dut, fmt.Sprintf(qosMapTC, index, index))
+		time.Sleep(time.Second)
+		runCliCommand(t, dut, fmt.Sprintf(qosCfgTargetGroup, index, fmt.Sprintf("target-group-%s", queue)))
+		time.Sleep(time.Second)
+	}
+
+}
+
+func buildCliSetRequest(config string) *gpb.SetRequest {
+	gpbSetRequest := &gpb.SetRequest{
+		Update: []*gpb.Update{
+			{
+				Path: &gpb.Path{
+					Origin: "cli",
+					Elem:   []*gpb.PathElem{},
+				},
+				Val: &gpb.TypedValue{
+					Value: &gpb.TypedValue_AsciiVal{
+						AsciiVal: config,
+					},
+				},
+			},
+		},
+	}
+	return gpbSetRequest
+}
+
+func configureTcamProfile(t *testing.T, dut *ondatra.DUTDevice) {
```

Comment:
move to cfgplugin, perhaps internal/cfgplugins/dut_initialize.go.   This function could be called NewHwInitialize and the vendor specific code should have a switch dut.Vendor statement.